service/src/main.rs

```diff
@@ -874,5 +884,10 @@ fn we_are_primary_validateer(
 ) -> Result<bool, Error> {
 	let enclave_count_of_previous_block =
 		node_api.enclave_count(Some(*register_enclave_xt_header.parent_hash()))?;
+	println!(
+		"enclave count is {} for previous block 0x{:?}",
+		enclave_count_of_previous_block,
+		register_enclave_xt_header.parent_hash()
+	);
 	Ok(enclave_count_of_previous_block == 0)
```

Comment:
I really don't get the logic of this. It's here for a while already: why so complicated?
why check that enclave count was zero in the previous block when we can check that it is 1 in the current block?
I didn't touch it beyond logging because we'll kick that out anyway and replace it with per-shard ShardStatus logic

Reply:
I have no idea actually.